core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/DriverComponent.scala

```diff
@@ -119,6 +120,48 @@ trait DriverComponent {
   def renameForHiding(recordCount: Long, name: String): String = {
     name + "_" + getSufficientlyRandomSuffix(recordCount)
   }
+
+  /**
+    * Converts a `Seq[(A, B)]` into a `Map[A, B]`, combining the values with a `Monoid[B]` in case of key conflicts.
+    *
+    * For example:
+    * {{{
+    * scala> groupPairs(Seq(("a", 1), ("b", 2), ("a", 3)))
+    * res0: Map[String,Int] = Map(b -> 2, a -> 4)
+    * }}}
+    * */
+  def groupPairs[A, B: Monoid](pairs: Seq[(A, B)]): Map[A, B] =
+    pairs.toList.foldMap { case (a, b) => Map(a -> b) }
+
+  /**
+    * Converts a `Seq[(A, F[B])]` into a `Map[A, F[B]]`, combining the values with the _universal_ monoid for type F.
+    * This can be useful for when using groupBy with slick aggregate functions such as max, min, avg, etc which return
+    * an Option.
+    *
+    * For example:
+    * {{{
+    * scala> case class Foo(i: Int)
+    * defined class Foo
+    *
+    * scala> groupPairs(Seq(("a", Foo(1).some), ("b", Foo(2).some), ("c", Foo(3).some)))
+    * << does not compile as there is no Monoid instance for Foo >>
+    *
+    * scala> groupPairsK(Seq(("a", Foo(1).some), ("b", Foo(2).some), ("c", Foo(3).some)))
+    * res9: Map[String,Option[Foo]] = Map(b -> Some(Foo(2)), a -> Some(Foo(1)), c -> Some(Foo(3)))
+    * }}}
+    */
+  def groupPairsK[F[_], A, B](pairs: Seq[(A, F[B])])(implicit M: MonoidK[F]): Map[A, F[B]] =
+    groupPairs(pairs)(M.algebra[B])
```

Comment:
A bunch of feedback here.

0. Check out `CollectionUtils` - seems like a good place for these to go.
1. I don't understand this.
2. Link to the [HTML version](http://typelevel.org/cats/typeclasses/semigroupk.html) of the documentation: it contains hugely important comment lines that aren't rendered inside GitHub!
3. sbt console gives me this when I try to define groupPairsK:

```
<pastie>:18: warning: higher-kinded type should be enabled
by making the implicit value scala.language.higherKinds visible.
This can be achieved by adding the import clause 'import scala.language.higherKinds'
or by setting the compiler option -language:higherKinds.
See the Scaladoc for value scala.language.higherKinds for a discussion
why the feature should be explicitly enabled.
```

4. `groupPairs` returns a map. You're then dereferencing the key defined by `M.algebra[B]`? I don't know what that does and I'd have thought that this would give you a result type of `F[B]`.
5. Your `groupPairsK` example is equivalent to `toMap`, which doesn't help me understand. Perhaps

```
scala> groupPairsK(Seq(("a", Foo(1).some), ("b", Foo(2).some), ("b", Foo(3).some)))
res0: Map[String,Option[Foo]] = Map(b -> Some(Foo(2)), a -> Some(Foo(1)))
```

would be more helpful, assuming I've read the Cats documentation (and therefore know that in the absence of a quantified `F` (= `Option`) means it'll just do `orElse`).

TLDR: I am :+1: on `groupPairs` and `groupTriples`, though they should go into `CollectionUtils`. I am :-1: on the K-versions; they rely on non-obvious behaviour (e.g. `orElse` for `Option`) that make them hard to understand.